src/components/limited-restoration-panel/LimitedRestorationPanel.vue

```diff
@@ -61,31 +70,37 @@ export default class LimitedRestorationPanel extends Mixins(DateMixin) {
   /** The validation rules for the Month. */
   private setMonthRules (): void {
     this.monthRules = [
-      (v:string) => !!v || 'Please enter the number of months',
-      (v:string) => (v > 0 && v <= 24) || 'This field must be between 1 to 24', // Keep month between 1 and 24
-      (v:string) => (v % 1 === 0) || 'Please enter a valid value' // Check if month is an integer
+      (v:string) => !!v || 'Must be between 1 to ' + this.maxNumberOfMonths,
+      (v:string) => (v > 0 && v <= this.maxNumberOfMonths) || 'Must be between 1 to ' + this.maxNumberOfMonths,
+      (v:string) => (v % 1 === 0) || 'Must be between 1 to ' + this.maxNumberOfMonths // Check if month is an integer
     ]
   }
 
+  /** Triggers the form validation. */
+  public validate (): void {
```

Comment:
This suggests you want the parent to call this function. I think it would be better to watch a `validate` prop. Eg,
https://github.com/bcgov/bcrs-shared-components/blob/78926fd0021ff0e782d8d4d302ec968e1134daf5/src/components/certify/Certify.vue#L147

Reply:
I don't get this. I'm calling this function whenever the number of months entered is changed. Each time it's changed, I check if the number is valid or not by calling this function. Hence, why would I watch a validate `prop`?

Reply:
It's basically like this here:
https://github.com/bcgov/bcrs-shared-components/blob/78926fd0021ff0e782d8d4d302ec968e1134daf5/src/components/approval-type/ApprovalType.vue#L149

Reply:
Maybe I misunderstood. I saw that it's public (for external use, like a parent asking a child to do something) instead of private (for internal use only). I assumed it would be called to validate the current form (like some other components do).

It seems that you only call it from 1 place. I think it would be less confusing if you moved its contents into `onMonthsChanged()`.

Reply:
> It's basically like this here:

Yes, I see that now. It's not used as I expected (sorry).

Reply:
Great idea actually. I will do that.

Reply:
Thanks.

To explain where I was coming from: some component are told when to validate the form (eg, when step-wise or filing-wise validation is enabled). One way that could be done is with a public function (as above) that the parent could call, like this: `this.$refs.mySubcomponent.validate()`, but this isn't good practice (except for form components, where we often call validate or reset on them). Another way is to have a prop, `validate`, that the component watches and reacts to. This may not be good practice either, but many of our components do that so at least it's consistent :)

Reply:
Hmm okay! Got it. I'll keep that in mind moving forward.